src/popover/popover.directive.ts

```diff
@@ -95,6 +96,12 @@ export class PopoverContainer {
 		"left" | "left-bottom" | "left-top" |
 		"right" | "right-bottom" | "right-top" = "bottom";
 
+	private changeDetectorRef: ChangeDetectorRef;
+
+	constructor(ref?: ChangeDetectorRef) {
+		this.changeDetectorRef = ref;
+	}
```

Comment:
Why do we need to make the dependency injection optional?

```suggestion
	constructor(private changeDetectorRef: ChangeDetectorRef) {}
```

Reply:
I could make it required for sure. All component that will extend this will then have to pass it. I left the option to make it optional if some component didn't need "want" to implement OnPush. On Angular 16 we could solve this in a nicer way by just using DI with the inject function not poluting the constructor. What do you think is the best approach here?

Reply:
I think it makes sense to add it as a required dependency, Ill be injecting additional dependencies sometimes next week that will be required when I integrate floating-ui with popover.